### PR TITLE
Release v0.2.10: harden objective sequential capture memory safety

### PR DESCRIPTION
**Pull request by xmarre**
## Summary

Prepare and release v0.2.10 by hardening Objective Sequential Capture against host/CUDA memory pressure and containing recoverable research-capture failures so they do not abort unrelated ComfyUI output nodes.

The bounded analysis transform, R/A/B semantics, objective metric profile/schema, forecasting behavior, sampler schedules, production model-aware defaults, and validated generic-correction default remain unchanged.

## Symptom and root cause

The reported failure occurred after generation when the Objective Sequential Capture path received decoded media, before the expected `stored R` completion. In the full workflow, an exception from this research node could also prevent unrelated output nodes such as Video Combine from running.

The exact historical allocation that exhausted the user's WSL/system headroom cannot be proven from the old log alone. The old path had confirmed memory-safety violations at that boundary:

- it allocated the full retained analysis destination while the decoded source was still live without checking current host/CUDA headroom;
- elementwise `torch.isfinite(chunk)` allocated a boolean tensor proportional to the source chunk;
- the fixed four-frame workspace scaled with source resolution;
- CPU float32 audio staging could alias the upstream waveform owner;
- duplicate/topology failures were discovered after staging;
- the bounded evaluator materialized reference, legacy, and candidate float32 chunks together.

For 192 float32 RGB frames at 960x736, the decoded source alone is about 1.516 GiB and the retained bounded video adds about 0.422 GiB before model/decoder/cache/process memory. The violated invariant was that retained media were bounded while staging admission and peak incremental workspace were not bounded against available memory.

## Memory-safety implementation

- Adds dependency-optional RSS/available-host telemetry with `psutil`, `/proc`, stdlib, and Windows fallbacks.
- Reports CUDA allocated/reserved/free memory when the source uses CUDA.
- Computes preflight requirements before retained destination allocation, including source media already live, new bounded VIDEO/AUDIO destinations, existing pending captures, staging workspace, and 512 MiB host / 256 MiB CUDA safety margins.
- Rejects known-unsafe captures before staging.
- Keeps the 4 GiB retained-analysis limit and applies a 6 GiB conservative incremental guard when host telemetry is unavailable.
- Derives VIDEO chunk size from a 64 MiB workspace target, capped at four frames; the ~0.7 MP case drops from four frames to three.
- Stages AUDIO into an independently owned CPU float32 tensor in bounded sample chunks.
- Replaces source-sized finite masks with `aminmax` plus a two-scalar finite/range check.
- Uses one owned NCHW work buffer, clamps in place, resizes in float32 only when needed, and releases intermediates before the next allocation.
- Serializes preflight, staging, and pending insertion so concurrent captures cannot admit against stale pending-state accounting.
- Performs duplicate role / topology / seed / config validation before staging.
- Processes legacy and candidate VIDEO chunks sequentially against one reference chunk in the bounded evaluator.
- Explicitly releases retained tensors on reset, eviction, completed-triad success, and completed-triad failure.

## Workflow failure semantics

Recoverable Objective Sequential Capture failures now return a status tuple rather than raising through the graph. Validation failures, incompatible/duplicate roles, preflight rejection, staging/evaluator/persistence failures, and unexpected non-resource-exhaustion exceptions are contained and logged. Earlier accepted R/A roles remain intact when a later role is rejected before the triad completes.

Actual resource exhaustion remains an exception. `MemoryError`, `torch.cuda.OutOfMemoryError`, and non-objective exceptions that report an out-of-memory condition are re-raised so ComfyUI/PyTorch can execute their normal OOM recovery path. `KeyboardInterrupt` is also not swallowed.

## Review fix

CodeRabbit identified one real peak-accounting bug in the no-resize branch: `work` remained aliased after ownership moved to `resized`, allowing the next chunk allocation to overlap the previous workspace. Commit `3bdf1e2` drops the stale binding immediately and expands numeric-equivalence coverage across float16, bfloat16, float32, and float64. The review thread is resolved.

## v0.2.10 release metadata

- `pyproject.toml` advances to `0.2.10`.
- `RELEASE_NOTES.md` is rewritten for this release.
- replay-calibration and generic-correction source-checkout fallback provenance versions advance to `0.2.10`.
- no manual tag or GitHub release is created by this PR; after merge, the repository's existing release workflow publishes only after the tested `main` commit succeeds.
- the Comfy Registry publish workflow is triggered by the `pyproject.toml` change on `main`.

## Validation

Final PR head: `10ca85be12c6a7a0604e243a74d27a8d8adbe2cb`.

GitHub Actions `tests` run #312 (`31848135647`) passed on all four pinned ComfyUI revisions:

- `27bca654eb9a70237d93f56a6ea336ab55f8925d`
- `0dd9b154a1654fc699dcdc3af066c7cce096045a`
- `e377e263049f9338b4d12a3dd417b36ae62948ff`
- `5599a05fea715cb2aff11f30f5b06e16d0dfa0c4`

Every matrix job passed wheel build, native-H3 forecaster smoke, scoped Ruff, `compileall`, and the full native test suite. The reviewed `27bca654` job reports **503 passed, 5 CUDA-only skipped**. The release wheel built successfully as `comfyui_spectrum_minimax_h3-0.2.10-py3-none-any.whl`.

Focused coverage includes deterministic staging parity, bounded workspace accounting, rejection before destination allocation, missing-telemetry fallback, duplicate/topology checks before staging, recoverable-failure containment, OOM propagation, source ownership, reset/eviction release, completed-triad teardown, and optional CUDA staging.

## Remaining uncertainty

Preflight is admission control, not an atomic reservation of system or device memory. A true OOM can still occur if WSL/Windows commit availability changes after preflight, another process allocates concurrently, allocator/backend scratch exceeds the estimate, or an accelerator does not expose usable free-memory telemetry. Those failures remain visible and intentionally propagate.

### CODE REVIEW

File: comfyui_spectrum_h3/objective_media_nodes.py [hunk 1]
```diff
@@ -253,57 +413,653 @@ def _source_video_metadata(video: torch.Tensor) -> dict[str, Any]:
...
+                or source_metadata["width"] != target_width
+            ):
+                resized = F.interpolate(
+                    work,
+                    size=(target_height, target_width),
+                    mode="area",
+                )
+                del work
+            else:
+                resized = work
```
[the remark this answers is not in the document]
Comment (xmarre) [reply comfyui_spectrum_h3/objective_media_nodes.py:969]: @coderabbitai Fixed in 3bdf1e2. The no-resize branch now drops the `work` name immediately after transferring ownership to `resized`, so `del resized` releases the chunk workspace before the next iteration allocates another one. I also expanded the numeric-equivalence regression across float16, bfloat16, float32, and float64 as requested in the nitpick. Validation: 31 focused tests passed (1 expected CUDA skip); full suite 499 passed (5 expected CUDA skips); CI Ruff/compileall and `git diff --check` passed.

**Comment by xmarre** [comment]
v0.2.10 release finalization complete.

- Squash-merged PR #53 to `main` as `59cb4649c298d8f77edfa018ffcbccb4fc8ff7b5`.
- Post-merge `tests` run #313 (`31848298563`) passed on the full four-revision ComfyUI matrix.
- Comfy Registry publish run #19 (`31848298596`) completed successfully; the `Publish Custom Node` step is green.
- Release workflow run #304 (`31848374970`) completed successfully after the tested `main` commit.
- GitHub release `v0.2.10` is published against the merge SHA with both expected assets:
  - `ComfyUI-Spectrum-MiniMax-H3-v0.2.10.zip`
  - `SHA256SUMS`
- Release ZIP digest reported by GitHub: `sha256:c955a3057a79adab62cce1b12a5d775e82085292d93dfc9c0604e79836c4528a`.
- `main` now reports package version `0.2.10`.

No unresolved review threads remain.
